Apps/W1/HybridSL/app/src/codeunits/CloudMigration.Codeunit.al

```diff
@@ -0,0 +1,326 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+namespace Microsoft.DataMigration.SL;
+
+using Microsoft.DataMigration;
+using Microsoft.Utilities;
+using System.Integration;
+using Microsoft.Finance.Dimension;
+using Microsoft.Finance.GeneralLedger.Account;
+using Microsoft.Sales.Customer;
+using Microsoft.Purchases.Vendor;
+using Microsoft.Inventory.Item;
+
+codeunit 42001 "SL Cloud Migration"
+{
+    Access = Internal;
+    TableNo = "Hybrid Replication Summary";
+    trigger OnRun();
+    var
+        HybridCompanyStatus: Record "Hybrid Company Status";
+        HybridSLManagement: Codeunit "SL Hybrid Management";
+        SLHelperFunctions: Codeunit "SL Helper Functions";
+        SLMigrationErrorHandler: Codeunit "SL Migration Error Handler";
+        HybridHandleSLUpgradeError: Codeunit "SL Hybrid Handle Upgrade Error";
+        SLPopulateAccounts: Codeunit "SL Populate Accounts";
+        slPopulateFiscalPeriods: Codeunit "SL Populate Fiscal Periods";
+        slDimensions: Codeunit "SL Dimensions";
+        Success: Boolean;
+    begin
+        SLMigrationErrorHandler.ClearErrorOccurred();
+        ClearLastError();
+        slPopulateFiscalPeriods.CreateFiscalPeriodsFromGLSetup();
+        slDimensions.InsertSLSegmentsForDimensionSets();
+        slDimensions.CreateSLCodes();
+        SLPopulateAccounts.PopulateSLAccounts();
```

Comment:
We should not run any logic outise of the OnUpgradeSLCompany. Why is this code outside?

Reply:
It is important that all logic runs within the event so we get error handling and telemetry done correctly. It will also ensure that all companies are upgraded in case any error is thrown.

Reply:
This logic has been moved to within OnUpgradeSLCompany.